### ISSUE TITLE
Fly South First on Box test

### ISSUE THREAD

**Issue by JaneClelandHuang** [DroneResponse/hardware-tests#19]
We need to fly South first for our Box test at Bremen.
I believe that all we have to do is to change this line:
https://github.com/DroneResponse/hardware-tests/blob/2242f9c32bd8d718f5a768156e34ce92f18bd4ec/nodes/box.py#L61

We would change it from:
return [start, north, east, south, west, north, end]

to:
return [start, south, east, north, west, south, end]

@murphym18  please confirm.
@tawfiq1200  (fyi)

**Comment by tawfiq1200** [comment]
When I tried to run the box test in Gazebo using the code in the master
branch, I could not run it due to something related to the Rcin, Rcin is
missing and the drone did not fly. The indoor_sensor test ran successfully.

**Comment by JaneClelandHuang** [comment]
I don't think we need to change this
https://github.com/DroneResponse/hardware-tests/blob/2242f9c32bd8d718f5a768156e34ce92f18bd4ec/nodes/box.py#L70

It tells it to move West before starting the cube test, and I think that is fine.  There will be plenty of space.  However, we might want it to also move a little south so maybe:
center = current_pos.move_ned(0.0, -25.0, 0.0)
becomes:
center = current_pos.move_ned(-5.0, -25.0, 0.0)
??

**Comment by JaneClelandHuang** [comment]
@tawfiq1200  that means it is looking for the RC, which makes sense.  It means we can't run this in the simulator without the RC connected and on.  I'm not going to fly this test until @murphym18 agrees with the proposed changes (and maybe kindly applies them).

**Comment by murphym18** [comment]
To change the box test, so it flies south we need to change
https://github.com/DroneResponse/hardware-tests/blob/2242f9c32bd8d718f5a768156e34ce92f18bd4ec/nodes/box.py#L70

The box coordinates are calculated relative to the center of the box. This line controls where the center of the box lives. We'd need to change it to:

```python
 center = current_pos.move_ned(-25.0, 0.0, 0.0) 
```

I will push the change

**Comment by JaneClelandHuang** [comment]
Thanks Michael.

**Comment by murphym18** [comment]
I uploaded a couple videos showing the ~~westward~~ southward box test:
- https://drive.google.com/file/d/1zg6lzQ0uzVnaHazTo7Q2QCRHtoXTqNKd/view?usp=sharing
- https://drive.google.com/file/d/1ConIhGLkLK-er7DyCtozQcU1AvvpXk_k/view?usp=sharing

I tested it in a few ways:
- I did an RC takeover mid-flight
- I gave it a realistic LLA for the South Bend RC club (using [the USGS altitude lookup tool](https://apps.nationalmap.gov/elevation/##bottom))
- I tested it with a takeoff alt of 7 meters and 2.5 meters
- I tested it with the drone facing south and east

It worked as expected in all cases.

Note: the drone flies to the Northern corner first. So it flies 25 meters south, then it flies back 5 meters.

**Comment by murphym18** [comment]
@tawfiq1200 to test this in the simulator I have to change the code slightly.

I made this patch file with the changes: [simulator.patch](https://drive.google.com/file/d/1xatGjRr79kiZzAXeUf-94ano0BzGd2Wg/view?usp=sharing)

Be careful with this patch. It bypasses a safety check and skips the "user ready to start" procedure where the pilot signals its time to start by operating the RC controls. 

To apply the patch:
```bash
git apply simulator.patch
```

**Comment by tawfiq1200** [comment]
@murphym18, I can see a path in the patch file, do I need to download it to
a specific location before applying the patch?

**Comment by murphym18** [comment]
The `simulator.patch` argument is a path. The example above assumes that the file is in the working-directory. But the file could be somewhere else too. For example:

```bash
cd hardware-tests
git apply ~/Downloads/simulator.patch
```

**Comment by murphym18** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Physical flight "unit" tests](https://trello.com/c/So9RcCKY/11-physical-flight-unit-tests)

**Comment by murphym18** [comment]
This was merged in fa0ebb7ceff6d8ae6a1495b9a83de82114b220bb
